### PR TITLE
Add support for Neptune Analytics as a Vector Store

### PR DESCRIPTION
**Pull request by bechbd**
# Description

Added support for Neptune Analytics as a Vector Store

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ X] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X ] Added new unit/integration tests
- [ X] Added new notebook (that tests end-to-end)
- [ X] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [X ] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation
- [X ] I have added Google Colab support for the newly added notebooks.
- [X ] My changes generate no new warnings
- [X ] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing unit tests pass locally with my changes
- [X ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-neptune/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+description = "llama-index vector_stores neptune integration"
+license = "MIT"
+name = "llama-index-vector-stores-neptune"
+packages = [{include = "llama_index/"}]
+readme = "README.md"
+version = "0.1.0"
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.0"
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-neptune/pyproject.toml:37]: Are there any other dependencies needed here? I forget if boto3 is included in `llama-index-core` or not. If not, it should be added here

**Review by logan-markewich — COMMENTED** [review]
LGTM, just a small nit about deps

File: llama-index-integrations/vector_stores/llama-index-vector-stores-neptune/pyproject.toml (hunk 1 above)
Comment (bechbd) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-neptune/pyproject.toml:37]: Nice catch, boto3 is not in `llama-index-core`, I've added it now.

**Comment by bechbd** [comment]
Thanks for the review, I believe that I have fixed all the issues.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**